### PR TITLE
feat: (re)adding sentry to public site

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

- [x] This change addresses the issue in full

## Description
This readds Sentry to the public site so we can keep track of the errors that are occurring and have been non-replicable 

## How Can This Be Tested/Reviewed?
Make sure you update your .env file to contain the new variable `SENTY_ORG`
Head to http://localhost:3000/sentry-example-page
and press the throw error button (if you have ad-block turned on you may need to turn it off)

head over to sentry and verify the error you threw is showing up there 

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
@YazeedLoonat Do you have a sense of if this has any impact on performance?

File: sites/public/.env.template [hunk 1]
```diff
@@ -13,6 +13,7 @@ CLOUDINARY_CLOUD_NAME=exygy
 CACHE_REVALIDATE=60
 NEW_RELIC_APP_NAME=Bloom Public
 NEW_RELIC_LICENSE_KEY=
+SENTY_ORG=
```
Comment (ludtkemorgan) [inline sites/public/.env.template:16]: minor, this should be "SENTRY"

File: sites/public/sentry.client.config.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// This file configures the initialization of Sentry on the client.
+// The config you add here will be used whenever a users loads a page in their browser.
+// https://docs.sentry.io/platforms/javascript/guides/nextjs/
+
+import * as Sentry from "@sentry/nextjs"
+
+Sentry.init({
+  dsn: "https://43432710ba2f4869a08318d2ad72028e@o1021204.ingest.sentry.io/5987038",
```
Comment (ludtkemorgan) [inline sites/public/sentry.client.config.ts:8]: Where does this URL come from? Is it something that should be secret?

**Review by ludtkemorgan — APPROVED** [review]
This looks good. Just minor questions/comments

File: sites/public/sentry.client.config.ts (hunk 2 above)
Comment (YazeedLoonat) [reply sites/public/sentry.client.config.ts:8]: Hey @ludtkemorgan thats the DSN and from sentry docs: https://docs.sentry.io/product/sentry-basics/dsn-explainer/#dsn-utilization

its publicly available. In Sentry we can set an allowlist of domains that can send a request our way, which is how we can control what sites' data gets into sentry

File: sites/public/.env.template (hunk 1 above)
Comment (YazeedLoonat) [reply sites/public/.env.template:16]: nice catch ty ty

**Comment by YazeedLoonat** [comment]
> @YazeedLoonat Do you have a sense of if this has any impact on performance?

Hey @emilyjablonski I think thats a bit tricky to know from just testing off of local, from what I can tell:

1. this does impact build time (the time it takes to run `yarn build`)
2. this hasn't impacted the actual performance of the application. Effectively its 1 post request being sent to sentry so I think this is relatively safe in terms of performance, but once we have it in a hosted env I think we can get a better sense of it

**Review by emilyjablonski — APPROVED** [review]

File: sites/public/src/pages/sentry-example-page.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      <main
+        style={{
+          minHeight: "100vh",
+          display: "flex",
+          flexDirection: "column",
+          justifyContent: "center",
+          alignItems: "center",
+        }}
+      >
+        <h1 style={{ fontSize: "4rem", margin: "14px 0" }}>
```
[the remark this answers is not in the document]
Comment (YazeedLoonat) [reply sites/public/src/pages/sentry-example-page.js:21]: this is just a test page for sentry we can safely ignore

**Pull request merged**